[charts] Improvement and docs on axis (#8654)
mui/mui-x: docs/data/charts/axis/axis.md

---
product: charts
title: Charts - Axis
---

# Charts - Axis

<p class="description">Axis provides associate values to element positions</p>

Axis are used in the following charts: `<LineChart />`, `<BarChart />`, `<ScatterChart />`.

## Defining axis

Like your data, axis definition play a central role in the charts rendering.
It's responsible of the mapping between your data and elements positions.

> You can define custom axis by using `xAxis` and `yAxis` props.

Author: I start wondering if it should be call `xAxes` because it's an array of axes. But it feels weird because I do not remember to have use "axes" in libraries I used for charting

> You can define custom axis by using `xAxis` and `yAxis` props.

Reviewer: Yeah, the prop name in plural form does indeed sound strange. If you could investigate what plural form are other libraries using that would be best 👌 🤔

Those props expect an array of objects.

Here is a demonstration with two lines with the same data.
But one uses a linear, and the other a log axis.

Each axis definition is identified by its property `id`.
And series specify the axis they use with `xAxisKey` and `yAxisKey` properties.

{{"demo": "ScaleExample.js", "bg": "inline"}}

### Axis type

> The axis type is specified by its property `scaleName` which expect one of the following values:

Author: `scaleType`?

> The axis type is specified by its property `scaleName` which expect one of the following values:

Reviewer: Oh yeah, that's way better! 👍

- `'band'`: Split the axis in equal band. Mostly used for bar charts.
- `'linear'`, `'log'`, `'sqrt'`: Map numerical values to the space available for the chart. `'linear'` is the default behavior.
- `'time'`, `'utc'`: Map javascript `Date()` object to the space available for the chart.

### Axis data

The axis definition object also includes a `data` properties.
Which expects an array of value coherent with the `scaleName`:

- For `'linear'`, `'log'`, or `'sqrt'` it should contain numerical values
- For `'time'` or `'utc'` it should contain `Date()` objects
- For `'band'` it can contain `string`, or numerical values

Moreover, some series types require specific axis types:

- line plots require a x-axis to have `data` provided
- bar blots require a x-axis with `scaleName='band'` and some `data` provided.

### Axis sub domain

By default, the axis domain is computed such that all your data are visible.

> To show a specific range of values, you can provide properties `min` and/or `max` to axis definitions.

Reviewer: ```suggestion
To show a specific range of values, you can provide properties `min` and/or `max` to the axis definition.
```

```js
xAxis={[
  {
    id: 'axisId',
    min: 10,
    max: 50,
  }
]}
```

{{"demo": "MinMaxExample.js", "bg": "inline"}}

## Axis customization

> Beside the impact of axis definition on the rendered chart, you can also customize the axis rendering.

Reviewer: ```suggestion
You can further customize the axis rendering besides the axis definition.
```

### Position

> To do that, charts components provide 4 props names `topAxis`, `rightAxis`, `bottomAxis`, and `leftAxis` allowsing to define the 4 axis of the chart.

Reviewer: ```suggestion
Charts components provide 4 props: `topAxis`, `rightAxis`, `bottomAxis`, and `leftAxis` allowing to define the 4 axes of the chart.
```
